cwms_radar_api/src/main/java/cwms/radar/api/LocationController.java

```diff
@@ -157,22 +183,209 @@ public void getOne(Context ctx, String locationCode) {
+                    required = true),
+            description = "Update CWMS Location",
+            method = HttpMethod.PATCH,
+            path = "/locations",
+            tags = {"Locations"}
+    )
     @Override
-    public void update(Context ctx, String locationCode) {
-        ctx.status(HttpServletResponse.SC_NOT_IMPLEMENTED).json(RadarError.notImplemented());
+    public void update(Context ctx, @NotNull String locationId)
+    {
+        updateRequest.mark();
+        String deserializationErrorMsg = null;
+        try(final Timer.Context timeContext = updateRequestTime.time();
+            DSLContext dsl = getDslContext(ctx))
+        {
+            LocationsDao locationsDao = new LocationsDaoImpl(dsl);
+            String acceptHeader = ctx.header(Header.ACCEPT);
+            String formatHeader = (acceptHeader != null) ? acceptHeader : Formats.JSON;
+            ContentType contentType = Formats.parseHeader(formatHeader);
+            if(contentType == null)
+            {
+                throw new FormattingException("Format header could not be parsed");
+            }
+            if(contentType.getType().equals(Formats.JSON))
+            {
+                ObjectMapper om = new ObjectMapper();
+                om.registerModule(new JavaTimeModule());
+                Location locationFromBody = deserializeJSONLocation(ctx.body());
+                //getLocation will throw an error if location does not exist
+                Location existingLocation = locationsDao.getLocation(locationId, UnitSystem.EN.getValue(), locationFromBody.getOfficeId());
+                //only store (update) if location does exist
+                Location updatedLocation = getUpdatedLocation(existingLocation, locationFromBody);
+                if(!updatedLocation.getName().equalsIgnoreCase(existingLocation.getName())) //if name changed then delete location with old name
+                {
+                    locationsDao.renameLocation(locationId, updatedLocation);
+                    ctx.status(HttpServletResponse.SC_ACCEPTED).json("Updated and renamed Location");
+                }
+                else
+                {
+                    locationsDao.storeLocation(updatedLocation);
+                    ctx.status(HttpServletResponse.SC_ACCEPTED).json("Updated Location");
+                }
+
+            }
+        }
+        catch(IOException ex)
+        {
+            RadarError re = new RadarError("Failed to process request: " + ex.getLocalizedMessage());
+            logger.log(Level.SEVERE, re.toString(), ex);
+            ctx.status(HttpServletResponse.SC_INTERNAL_SERVER_ERROR).json(re);
+        }
+
     }
 
-    @OpenApi(ignore = true)
+    @OpenApi(
+            queryParams = {
+                    @OpenApiParam(name = "office", description = "Specifies the owning office of the location whose data is to be deleted. If this field is not specified, matching location information will be deleted from all offices.")
```

Comment:
Aren't there options in the location delete to decide if the operation should cascade to dependent objects? or are we intentionally leaving that out right now for initial safety?

Reply:
We can add the cascade later when we've better decided how to handle that.